woocommerce/woocommerce-android
# Store Creation: Reintroduce store profiler steps and send info during free trial creation

Part of: #8824


This is the first part of the task to re-introduce store profilers and send the data during free trial creation.

@jaclync on her research on pe5sF9-1pZ-p2#comment-2025 shared that the free trial endpoint now accepts profiler data to be submitted in the request body. The JSON structure should look like below:

```json
{
    "wpcom_woocommerce_onboarding": {
        "blogname": "test trial", // from the store name question
        "woocommerce_default_country": "TW", // from the country code question
        "woocommerce_onboarding_profile": {
            "industry": [
                {
                    "slug": "fashion-apparel-accessories", // The industry group, which we're updating the options as another subtask
                    "detail": "" // Only populated when it's a subcategory
                }
            ],
            "is_store_country_set": true,
            "selling_venues": "no"|"brick-mortar"|"other", // from the selling status question
            "other_platform": "amazon", // from the selling status followup question if the merchant is already selling online
            "skipped": true // seems to be always true, we can probably ignore this field
        }
    }
}
```

This PR:

1. Re-enables the store profiler flow that was previously disabled. Currently it's still hidden behind feature flag while the tasks on #8824 are being worked on.
2. Passes the data and adds a converter to add it into the free trial plan addition API call.

Please note that this PR just restores the previously existing site profile flow and options as-is. Any latest update/changes will be added in a separate PR.


### Description


### Testing instructions

1. Start a new store,
2. After entering store name, ensure the store profiler flow is displayed,
3. After completing the flow, ensure the country picker is displayed,
4. Continue forward to the site creation process and ensure the site is created.

Optionally, debug the app, put a breakpoint on line 131 of `SitePlanRestClient` and check the included `body` and see if it matches the JSON structure above.

### Video

[device-2023-04-18-155253.webm](https://user-images.githubusercontent.com/266376/232726236-230a6e66-514f-417e-9042-a42ca55ab56c.webm)
